test/pipelines/test_ray.py

```diff
@@ -14,18 +14,21 @@ def shutdown_ray():
     try:
         import ray
 
+        ray.serve.shutdown()
         ray.shutdown()
     except:
         pass
 
 
 @pytest.mark.integration
 @pytest.mark.parametrize("document_store_with_docs", ["elasticsearch"], indirect=True)
-def test_load_pipeline(document_store_with_docs):
+@pytest.mark.parametrize("serve_detached", [(True,), (False,)])
+def test_load_pipeline(document_store_with_docs, serve_detached):
     pipeline = RayPipeline.load_from_yaml(
         SAMPLES_PATH / "pipeline" / "ray.haystack-pipeline.yml",
         pipeline_name="ray_query_pipeline",
         ray_args={"num_cpus": 8},
+        serve_args={"detached": serve_detached},
```

Comment:
Minor: do you think we can test for this parameter to be actually passed to Ray Serve?

Reply:
That is an interesting question and the answer is not obvious. Meaning the answer is - MAYBE. :-)
Let me look into it and report back.

Reply:
Hi @ZanSara, I had to surface the ServeControllerClient to be able to test for this parameter to actually being passed to Ray Serve. See latest commit.
I hope now this is ready for merging - but if you see anything else missing, please let me know.